### PR TITLE
Course DTO

### PR DESCRIPTION
**Pull request by neoed**

### CODE REVIEW

**Comment by PES34** [comment]
description of commit is not clear

**Comment by PES34** [comment]
You make more changes than write in commit

File: src/main/java/com/inqoo/trainingservice/app/DTO/CategoryDTO.java [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
 public class CategoryDTO {
-    public final String name;
-    public final String description;
+    private final String name;
+    private final String description;
+    private final UUID uuidCategory;
 
-    public CategoryDTO(String name, String description) {
+    public CategoryDTO(String name, String description, UUID uuidCategory) {
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/DTO/CategoryDTO.java:10]: How about categoryUUID?

File: src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private final Long duration;
+    private BigDecimal price;
+
+    public CourseDTO(String name, String description,  Long duration, BigDecimal price, UUID uuidCourse) {
+        this.name = name;
+        this.description = description;
+        this.duration = duration;
+        this.price = price;
+        this.uuidCourse = uuidCourse;
+
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java:19]: empty line

File: src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    private final UUID uuidCourse;
+    private final Long duration;
+    private BigDecimal price;
+
+    public CourseDTO(String name, String description,  Long duration, BigDecimal price, UUID uuidCourse) {
+        this.name = name;
+        this.description = description;
+        this.duration = duration;
+        this.price = price;
+        this.uuidCourse = uuidCourse;
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java:18]: how about courseUUID?

File: src/main/java/com/inqoo/trainingservice/app/models/Category.java [hunk 4]
```diff
@@ -13,14 +15,17 @@
     private String name;
     @Column(length = 300)
     private String description;
+    @Column
+    private UUID uuidCategory;
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/models/Category.java:19]: how about categoryUUID?

File: src/main/java/com/inqoo/trainingservice/app/models/Course.java [hunk 5]
```diff
@@ -14,12 +15,15 @@
     private String description;
     private Long duration;
     private BigDecimal price;
+    @Column
+    private UUID uuidCourse;
     
-    public Course(String name, String description, Long duration, BigDecimal price) {
+    public Course(String name, String description, Long duration, BigDecimal price, UUID uuidCourse) {
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/models/Course.java:21]: maybe courseUUID?

File: src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java [hunk 6]
```diff
@@ -31,17 +30,15 @@ public void shouldReturnListOfCategory() {
         //given
         CategoryDTO category1 = new CategoryDTO(
                 "JavaBasic",
-                "Podstawy Javy");
+                "Podstawy Javy", UUID.randomUUID());
```
Comment (adamkulawik) [inline src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java:33]: format it, please

Comment (adamkulawik) [reply src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java:33]: and all other places here too

File: src/test/java/com/inqoo/trainingservice/app/service/CourseServiceIT.java [hunk 7]
```diff
@@ -56,18 +65,18 @@ public void shouldReturnListOfCourses() {
...
+        categoryService.saveNewCategory(categoryConverter.dtoToEntity(category));
         Subcategory subcategory = new Subcategory("Spring", "Kurs Spring");
         subcategoryService.saveNewSubcategory(subcategory, category.getName());
-        Course course = new Course(
+        CourseDTO course = new CourseDTO(
                 "Spring Boot w Javie",
                 "Kurs na temat Spring Boot w Javie",
                 2L,
-                BigDecimal.valueOf(2000));
+                BigDecimal.valueOf(2000), UUID.randomUUID());
```
Comment (adamkulawik) [inline src/test/java/com/inqoo/trainingservice/app/service/CourseServiceIT.java:76]: format it

File: src/main/java/com/inqoo/trainingservice/app/models/Course.java (hunk 5 above)
Comment (neoed) [reply src/main/java/com/inqoo/trainingservice/app/models/Course.java:21]: done

File: src/main/java/com/inqoo/trainingservice/app/models/Category.java (hunk 4 above)
Comment (neoed) [reply src/main/java/com/inqoo/trainingservice/app/models/Category.java:19]: done

File: src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java (hunk 3 above)
Comment (neoed) [reply src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java:18]: done

File: src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java (hunk 2 above)
Comment (neoed) [reply src/main/java/com/inqoo/trainingservice/app/DTO/CourseDTO.java:19]: done

File: src/main/java/com/inqoo/trainingservice/app/DTO/CategoryDTO.java (hunk 1 above)
Comment (neoed) [reply src/main/java/com/inqoo/trainingservice/app/DTO/CategoryDTO.java:10]: done

File: src/test/java/com/inqoo/trainingservice/app/service/CourseServiceIT.java (hunk 7 above)
Comment (neoed) [reply src/test/java/com/inqoo/trainingservice/app/service/CourseServiceIT.java:88]: done

File: src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java (hunk 6 above)
Comment (neoed) [reply src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java]: done

**Review by adamkulawik — APPROVED** [review]

**Pull request merged**